chore: revamp the website

skyrpex opened the pull request.

skyrpex commented on src/commands.ts:
```diff
@@ -0,0 +1,86 @@
+  },
+  // {
+  //   name: "monada",
+  //   description: "The official Monada web client",
```
We can have some sort of monada binary file, with subcommands such as "why".

ainvoner replied:
@skyrpex we can have a Monada Object with all the relevant info about Monada and a "generic" Monada command like: Monada.xxxx, the string after "." will be the path to a specific Monada object key value.
f.e: Monada.why will return the value from :
```
const Monada = {
  why: "because we are awesome",
}
``` 
but than the "files" we have (which the user can "see" their content using "cat" command) are redundant

skyrpex replied:
As per what we talked privately, I implemented "monada culture", which asynchronously prints the CULTURE file. The file no longer exists as it is redundant.

skyrpex commented:
@eladb you can have a look at https://monadahq.github.io/monada-website/. Let me know if you'd like me to change or add something.

skyrpex commented:
I'll just add history navigation with the arrow keys real quick

noyfactor reviewed (commented):
General comment - this is a huge PR. In the future, I'd suggest splitting it into a few smaller ones to make the review easier

noyfactor commented on .vscode/settings.json:
```diff
@@ -0,0 +1,18 @@
+{
```
Can we make this global somehow?

skyrpex replied:
We can add an entry to internal-kb mentioning the recommended settings for VSCode (and maybe other editors?), and then remove it from this repo.

noyfactor commented on README.md:
```diff
@@ -1 +1,34 @@
\ No newline at end of file
+## Usage
+
+Those templates dependencies are maintained via [pnpm](https://pnpm.io) via `pnpm up -Lri`.
```
Why though? 🤔

skyrpex replied:
Oh, that comes from the base template I used to set everything up. I need to re-do this file completely.

I

noyfactor commented on package.json:
```diff
@@ -0,0 +1,27 @@
+  "version": "0.0.0",
+  "description": "",
+  "scripts": {
+    "dev": "vite",
```
cool, didn't know `vite` :) learned something today!

noyfactor commented on package.json:
```diff
@@ -0,0 +1,27 @@
+    "autoprefixer": "^10.4.7",
+    "postcss": "^8.4.14",
+    "prettier": "^2.7.0",
+    "tailwindcss": "^3.0.24",
```
tailwindcss 😍

skyrpex replied:
I like it very much, specially after trimming the bundled css :D

noyfactor commented on src/commands.ts:
```diff
@@ -0,0 +1,95 @@
+        //   .join("")}</tr></table>`,
+        `<div class="grid grid-cols-4 gap-4">${Object.keys(files)
+          .map((file) => `<div>${file}</div>`)
+          .join("")}</div>`,
```
Have to admit I'm a singlequotes type of person 🙈

skyrpex replied:
I personally don't care, it's just the default configuration for prettier. We can switch to single quotes :)

noyfactor commented on src/commands.ts:
```diff
@@ -0,0 +1,95 @@
+    },
+  },
+  {
+    name: "monada",
```
that's... somthing I'd be careful with :) If we don't get the API right it'll look lame rather than awesome

noyfactor replied:
Also still not working in the demo 🤔  not latest there?
<img width="287" alt="image" src="https://user-images.githubusercontent.com/5963602/174058953-6957ba4d-7138-42a2-a6d1-eae999a70719.png">

noyfactor replied:
I don't have anything concrete yet. I think we should:
1. Come up with one 😇 
2. Peer-review it internally
3. Let some (x>2) engs we appreciate fuss around with the website and see what they try to type in not knowing the api
<img width="552" alt="image" src="https://user-images.githubusercontent.com/5963602/174466905-5573b548-1356-41b7-9dfb-5da6140e9d88.png">

skyrpex replied:
Definitely, `--help` should be mandatory :D Will add it.

noyfactor commented on src/components/Terminal.vue:
```diff
@@ -0,0 +1,151 @@
+| .  . | ___  _ __   __ _  __| | __ _
+| |\\/| |/ _ \\| '_ \\ / _\` |/ _\` |/ _\` |
+| |  | | (_) | | | | (_| | (_| | (_| |
+\\_|  |_/\\___/|_| |_|\\__,_|\\__,_|\\__,_|.co</span>
```
I'd remove the `.co`, and also run this through browserstack to make sure it looks everywhere

skyrpex replied:
Works on Edge, Safari, Firefox, Opera and Chrome. It doesn't work on IE. 

That's because Vite, by default, only supports [Native ESM](https://caniuse.com/es6-module). We can add support for IE using the [legacy plugin](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy). This legacy plugin will generate some additional JS files that will be loaded only on older browsers, and will add some boilerplate to the index.html page to make it work (the file increases in size from 0.70 KiB to 1.87 KiB). I wouldn't bother supporting these legacy browsers, since the website very focused on developers. WDYT?

noyfactor replied:
Agreed, IMO we would just like to make sure it looks sharp on chrome, safari, ff

noyfactor commented on src/components/Terminal.vue:
```diff
@@ -0,0 +1,151 @@
+
+<template>
+  <div
+    class="w-full p-4 text-sm font-mono flex-1 whitespace-pre-wrap"
```
same comment re: `flex-1` :)

skyrpex replied:
I'm afraid I need it so it fills the remaining vertical space

noyfactor commented on src/App.vue:
```diff
@@ -0,0 +1,9 @@
+</script>
+
+<template>
+  <div class="flex-1 flex flex-col bg-slate-800 text-white cursor-text">
```
ooo can we use system preference for dark/light mode?

skyrpex replied:
Yeah! There's only two colors so it will be easy

skyrpex replied:
Done, now there's an adaptive theme and a command to switch between them.

skyrpex commented on src/App.vue:
```diff
@@ -0,0 +1,9 @@
+</script>
+
+<template>
+  <div class="flex-1 flex flex-col bg-slate-800 text-white cursor-text">
```
That's right. I did that so the cursor-text applies to the whole page, but I can specify it on the html or the body instead 👍🏻

skyrpex replied:
I still need some `flex-1` along with `flex flex-col` so my terminal component fills the entire screen vertically. It's necessary to add some re-focusing to the input field after the user clicks the screen.

Feel free to suggest a different way to make it work, or submit a PR if you prefer 👍🏻

eladb commented:
@skyrpex what's the best way to look at a live version of this?

skyrpex commented:
> @skyrpex what's the best way to look at a live version of this?

I'm manually deploying it to GitHub pages from time to time, [here](https://monadahq.github.io/monada-website/)

noyfactor commented:
Analytics 😱

eladb commented:
> Analytics 😱 

Good point. It's a good opportunity to select a good solution. 

Anyone has experience with that?

ainvoner commented:
for the current webpage (which will probably change once we will need to sell / promote our products to humans :)) we can use google analytics with its' framework specific packages (for Vue in our case).
since we are not interested only in the site traffic, we will be able to add custom events such as - who looked at the JOBS file...

skyrpex commented:
If we use cookie-enabled analytics, we'll have to add the infamous GDPR cookie consent, right?

There are alternatives that don't use cookies, like [plausible](https://plausible.io/) or [fathom](https://usefathom.com/pricing) (although not free). I've used plausible in the past.

I think that we can use cookie-less tracking with Google Analytics too 🤔

skyrpex commented:
Removed the JOBS file (we're not currently hiring), and also added cookie-less Google Analytics using [gtag](https://developers.google.com/analytics/devguides/collection/gtagjs#:~:text=The%20global%20site%20tag%20(gtag,non%2Dproduct%20specific)%20gtag.).

The GA project is https://analytics.google.com/analytics/web/#/p320119168/reports/intelligenthome?params=_u..pageSize%3D25%26_u..nav%3Dmaui. I'm looking for a setting to share it with y'all.

eladb reviewed (changes requested):
A few minor comments, most of them are generally in the spirit of minimalism.

Minimalistic design is something I want us to establish as a principal and it's basic foundation reduction. If we can achieve the same result with less, we will choose less.

Another benefit of minimalism is that there is less decisions to make about how things behave and operate.

The last part is that our bar for anything that is public facing is going to be very high. We want everything that is user facing to be intentional and have the highest polish and quality - this means that if we expose _less_ there is less finishing and polishing to do.

Again- probably way too much philosophical depth for this little thing but I'm hoping to use this opportunity to establish our high standards in this subject.

skyrpex commented on src/composables/commands.ts:
```diff
@@ -0,0 +1,127 @@
+      },
+    },
+    {
+      name: "monada",
```
Not really, in fact we didn't know what to do with it. Will remove it

skyrpex commented:
@eladb I went ahead and formatted the rest of the help messages to have the look & feel of Linux man, also the "command not found" message.

Regarding the analytics provider, @noyfactor suggested [amplitude](https://amplitude.com/), which I got working on another PR. I already created an account for Monada, but I think it needs additional configuration.

ainvoner commented on src/files.ts:
```diff
@@ -0,0 +1,3 @@
+export const files: Record<string, string> = {
+  "CONTACT-US": "ping@monada.co",
```
note to myself... also update the repo email https://github.com/organizations/monadahq/settings/profile

eladb replied:
I think we do have a ping@ address. @ainvoner Can you check?

eladb replied:
![image](https://user-images.githubusercontent.com/598796/175084846-505cea11-5f09-4a32-b070-978340cdcf67.png)

eladb replied:
There is a group in our Google Workspace called `ping@monada.co`. Let's leave it as "ping" please. I know it's geeky, but, well.. 

<img width="959" alt="Screen Shot 2022-06-22 at 22 10 29" src="https://user-images.githubusercontent.com/598796/175117433-e3ac6999-9675-40d5-813c-821a33754dd1.png">

skyrpex replied:
Ping it is

eladb reviewed (approved):
Looks very good!

For the roadmap: would be cool if I hit "TAB" it will show me the list of possible completions, like in the shell. This way, we don't really need `help`.

skyrpex commented:
> Looks very good!
> 
> For the roadmap: would be cool if I hit "TAB" it will show me the list of possible completions, like in the shell. This way, we don't really need `help`.

Tracked in issue https://github.com/monadahq/monada-website/issues/4
